### PR TITLE
Exclude for rules magic methods

### PR DESCRIPTION
**Pull request by matks**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Exclude for rules magic methods.
| Type?             | bug fix
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes half of https://github.com/PrestaShop/phpstan-prestashop/issues/15
| How to test?      | All tests pass
| Possible impacts? |

### CODE REVIEW

File: tests/Rules/UseTypeHintForNewMethodsRuleTest.php [hunk 1]
```diff
@@ -55,5 +55,6 @@ public function testRule(): void
         $this->analyse([$dataDirectory . 'MethodFooEThreeParametersWithIdentifierTypeHint.php'], []);
         $this->analyse([$dataDirectory . 'MethodFooGWithoutTypeHintButExcluded.php'], []);
         $this->analyse([$dataDirectory . 'MethodFooHWithoutTypeHintButExtendsAnExcludedClass.php'], []);
+        $this->analyse([$dataDirectory . 'JWithConstructAndGetSet.php'], []);
```
Comment (PierreRambaud) [inline tests/Rules/UseTypeHintForNewMethodsRuleTest.php]: Why `J`? :thinking:

File: tests/Rules/UseTypedReturnForNewMethodsRuleTest.php [hunk 2]
```diff
@@ -44,5 +44,6 @@ public function testRule(): void
         $this->analyse([$dataDirectory . 'MethodBarEWithoutReturnType.php'], []);
         $this->analyse([$dataDirectory . 'MethodBarFExtendsEWithoutReturnType.php'], []);
         $this->analyse([$dataDirectory . 'MethodFooGWithNullReturnType.php'], []);
+        $this->analyse([$dataDirectory . 'IWithConstructAndGetSet.php'], []);
```
Comment (PierreRambaud) [inline tests/Rules/UseTypedReturnForNewMethodsRuleTest.php]: Why `I`? :thinking:

File: tests/Rules/UseTypeHintForNewMethodsRuleTest.php (hunk 1 above)
Comment (matks) [reply tests/Rules/UseTypeHintForNewMethodsRuleTest.php]: Initially I had sample test files `A.php`, `B.php`, `C.php`.

Then I went for improved file names such as `MethodFooHWithoutTypeHintButExtendsAnExcludedClass.php` but I continue to add one alphabet letter in each file.

Indeed maybe it's completely useless 😁

File: tests/Rules/UseTypedReturnForNewMethodsRuleTest.php (hunk 2 above)
Comment (matks) [reply tests/Rules/UseTypedReturnForNewMethodsRuleTest.php]: Same as above 😄

File: tests/Rules/UseTypeHintForNewMethodsRuleTest.php (hunk 1 above)
Comment (PierreRambaud) [reply tests/Rules/UseTypeHintForNewMethodsRuleTest.php]: It's confusing :sweat_smile:

Comment (matks) [reply tests/Rules/UseTypeHintForNewMethodsRuleTest.php]: Replaced by "ClassWithConstructAndGetSet.php" in https://github.com/PrestaShop/phpstan-prestashop/pull/17/commits/5501859a03131b791d9cc8e70dc9d57c8f2d5e81

File: tests/Rules/UseTypedReturnForNewMethodsRuleTest.php (hunk 2 above)
Comment (matks) [reply tests/Rules/UseTypedReturnForNewMethodsRuleTest.php]: Replaced by "ClassWithConstructAndGetSet.php" in https://github.com/PrestaShop/phpstan-prestashop/pull/17/commits/5501859a03131b791d9cc8e70dc9d57c8f2d5e81

**Review by atomiix — APPROVED** [review]

**Pull request merged**